src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java

```diff
@@ -110,10 +108,12 @@ private void checkAndResetGameOnScore(ActivateEvent event, EntityRef entity) {
             EntityRef heldFlag = getHeldFlag(player);
             if (checkIfTeamScores(baseTeamComponent, heldFlag)) {
                 incrementScore(baseTeamComponent);
-                if (redScore < LASUtils.GOAL_SCORE && blackScore < LASUtils.GOAL_SCORE) {
-                    resetRound(baseTeamComponent, heldFlag);
-                } else {
-                    resetLevel(player, baseTeamComponent, heldFlag);
+                resetRound(baseTeamComponent, heldFlag);
+                if (redScore == LASUtils.GOAL_SCORE) {
```

Comment:
I'd like to see a `>=` here instead of `==` to harden this system against the (unlikely) event that the score increases by 2